src/Orleans.Core/Configuration/Options/GatewayOptions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Configuration
+{
+    public class GatewayOptions
+    {
+        /// <summary>
+        /// Default gateway list refresh period
+        /// </summary>
+        public static readonly TimeSpan DEFAULT_GATEWAY_LIST_REFRESH_PERIOD = TimeSpan.FromMinutes(1);
+        /// <summary>
+        /// Gateway list refresh period
+        /// </summary>
+        public TimeSpan GatewayListRefreshPeriod { get; set; } = DEFAULT_GATEWAY_LIST_REFRESH_PERIOD;
+        /// <summary>
+        /// Prefered gateway index
+        /// </summary>
+        public int PreferedGatewayIndex { get; set; } = -1;
```

Comment:
The special value of -1 should be documented, like: `Preferred gateway index or -1 to prefer no gateway`.